src/main/java/org/graylog/plugins/usagestatistics/collectors/ElasticsearchCollector.java

```diff
@@ -137,15 +146,24 @@ public ElasticsearchClusterStats getClusterStats() {
         );
     }
 
-    private Map<String, NodeInfo> fetchNodeInfos() {
-        ClusterAdminClient adminClient = this.client.admin().cluster();
-        NodesInfoResponse nodesInfoResponse = adminClient.nodesInfo(adminClient.prepareNodesInfo().request()).actionGet();
-        return nodesInfoResponse.getNodesMap();
-    }
-
-    private Map<String, NodeStats> fetchNodeStats() {
-        ClusterAdminClient adminClient = this.client.admin().cluster();
-        NodesStatsResponse nodesStatsResponse = adminClient.nodesStats(adminClient.prepareNodesStats().request()).actionGet();
-        return nodesStatsResponse.getNodesMap();
+    private JsonObject fetchNodeInfos() {
+        final NodesInfo.Builder requestBuilder = new NodesInfo.Builder()
+                .withHttp()
+                .withJvm()
+                .withNetwork()
+                .withOs()
+                .withPlugins()
+                .withProcess()
+                .withSettings()
+                .withThreadPool()
+                .withTransport();
+        try {
+            final JestResult result = jestClient.execute(requestBuilder.build());
+            return Optional.of(result.getJsonObject())
```

Comment:
This doesn't handle the case of an unsuccessful request and basically swallows any information which could be used to notice and investigate failures.